python/pynq/intf/pattern_generator.py

```diff
@@ -27,20 +27,29 @@
 #   OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
 #   ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-__author__ = "Yun Rock Qu"
-__copyright__ = "Copyright 2017, Xilinx"
-__email__ = "pynq_support@xilinx.com"
 
 import re
 import numpy as np
-from .intf_const import INTF_MICROBLAZE_BIN, IOSWITCH_PG_SELECT, \
-    PYNQZ1_DIO_SPECIFICATION, CMD_CONFIG_PG, CMD_ARM_PG, \
-    CMD_RUN, CMD_STOP
-from .intf import request_intf, _INTF
+from .intf_const import INTF_MICROBLAZE_BIN
+from .intf_const import MAX_NUM_PATTERN_SAMPLES
+from .intf_const import MAX_NUM_TRACE_SAMPLES
+from .intf_const import IOSWITCH_PG_SELECT
+from .intf_const import PYNQZ1_DIO_SPECIFICATION
+from .intf_const import CMD_CONFIG_PG
+from .intf_const import CMD_ARM_PG
+from .intf_const import CMD_RUN
+from .intf_const import CMD_STOP
+from .intf import request_intf
+from .intf import _INTF
 from .waveform import Waveform
 from .trace_analyzer import TraceAnalyzer
 
 
+__author__ = "Yun Rock Qu"
+__copyright__ = "Copyright 2017, Xilinx"
+__email__ = "pynq_support@xilinx.com"
+
+
```

Comment:
The variables have to be defined after import, otherwise the Pycharm will warn you that it is violating PEP8.